main.py

```diff
@@ -0,0 +1,77 @@
+from fastapi import FastAPI, Depends, HTTPException
+from sqlalchemy.orm import Session
+
+import schemas
+from crud import (
+    get_all_authors,
+    create_author,
+    get_author_detail,
+    get_all_books,
+    create_book,
+)
+from database import SessionLocal
+
+
+app = FastAPI()
+
+API_PREFIX = "/api/v1/"
+
+
+def get_db() -> Session:
+    db = SessionLocal()
+
+    try:
+        yield db
+    finally:
+        db.close()
+
+
+@app.get(API_PREFIX + "authors/", response_model=list[schemas.Author])
+def read_all_authors(
+        skip: int | None = None,
+        limit: int | None = None,
+        db: Session = Depends(get_db)
```

Comment:
It's a good practice to provide default values for parameters like `skip` and `limit` in pagination. However, the type hint `int | None` can be replaced with `Optional[int]` from the `typing` module, which is more readable and commonly used in the Python community.